torch/distributed/_tensor/dispatch.py

```diff
@@ -297,7 +297,11 @@ def unwrap_to_op_info(
                 else:
                     mesh = arg.device_mesh
             elif isinstance(arg, torch.Tensor):
-                if arg.ndim == 0 or self._allow_implicit_replication:
+                if (
+                    arg.ndim == 0
+                    or arg.numel() == 1
```

Comment:
I think: we might want to throw some warnings for the `arg.numel() == 1` case, where if we found such a thing, in addition to creating a replicated DTensor, we also need to throw warning to the user to let user either change their code to a scalar tensor, or create it with DTensor when in the distributed context

Reply:
> I think: we might want to throw some warnings for the `arg.numel() == 1` case, where if we found such a thing, in addition to creating a replicated DTensor, we also need to throw warning to the user to let user either change their code to a scalar tensor, or create it with DTensor when in the distributed context

Sounds good! Updated the PR to include the warning.